msmtools/util/sorted_schur.py

```diff
@@ -360,44 +276,29 @@ def sorted_schur(P, m, z='LM', method='brandts', tol_krylov=1e-16):
         warnings.warn("Sparse implementation is only avaiable for `method='krylov'`, densifying.")
         P = P.toarray()
 
+    # make sure we have enough eigenvalues to check for block splitting
+    n = P.shape[0]
+    if m < n:
+        k = m + 1
+    elif m == n:
+        k = m
+
+    # compute the sorted schur decomposition
     if method == 'brandts':
-        # Calculate the top m+1 eigenvalues and secure that you
-        # don't separate conjugate eigenvalues (corresponding to 2x2-block in R),
-        # if you take the dominant m eigenvalues to cluster the data.
-        #  _ = top_eigenvalues(P, m, z=z, tol=tol_krylov)
-   
-        # Make a Schur decomposition of P.
-        R, Q = schur(P, output='real')
-
-        # sort one more than requested
-        n = P.shape[0]
-        if m < n:
-            k = m + 1
-        elif m == n:
-            k = m
-        
-        # Sort the Schur matrix and vectors.
-        Q, R, ap = sort_real_schur(Q, R, z=z, b=k)
-
-        # comptue eigenvalues
-        T, _ = rsf2csf(R, Q)
-        eigenvalues = np.diag(T)[:k]
-
-        # check for splitting pairs of complex conjugates
-        if (m < n):
-            eigenval_in = eigenvalues[m - 1]
-            eigenval_out = eigenvalues[m]
-            if np.isclose(eigenval_in, np.conj(eigenval_out)):
-                raise ValueError(f'Clustering into {m} clusters will split conjugate eigenvalues. '
-                                 f'Request one cluster more or less. ')
-            Q, R, eigenvalues = Q[:, :m], R[:m, :m], eigenvalues
-
-        # Warnings
-        if np.any(np.array(ap) > 1.0):
-            warnings.warn("Reordering of Schur matrix was inaccurate.")
+        R, Q, eigenvalues = sorted_brandts_schur(P=P, k=k, z=z)
     elif method == 'krylov':
-        R, Q, eigenvalues, _ = sorted_krylov_schur(P, m, z=z, tol=tol_krylov)
+        R, Q, eigenvalues, _ = sorted_krylov_schur(P=P, k=k, z=z, tol=tol_krylov)
     else:
         raise ValueError(f"Unknown method `{method!r}`.")
+
+    # check for splitting pairs of complex conjugates
+    if (m < n):
+        if _check_conj_split(m, eigenvalues):
+            raise ValueError(f'Clustering into {m} clusters will split conjugate eigenvalues. '
```

Comment:
Yes, we could warn here instead of raising and further use Q[:, :m+1], R[:m+1, :m+1]...

Reply:
I opted for not doing this yet - I think it would be cleaner to do this in a separate PR.